creusot-contracts/src/logic/seq.rs

```diff
@@ -134,11 +134,19 @@ impl<T: ?Sized> Seq<T> {
         absurd
     }
 
+    #[trusted]
+    #[logic]
+    #[open(self)]
+    #[creusot::builtins = "prelude.prelude.Seq.cons_"]
```

Comment:
this uses "prelude.prelude.Seq.cons_", but push_back uses "seq.Seq.snoc", is there a reason for this difference?

Reply:
`seq.Seq.snoc` is from the why3 stdlib. Ideally we would also use `seq.Seq.cons` (also from the why3 stdlib), but the order of arguments is different from the one we want! (`self` has to be the first argument, right?)
Hence the `cons_` wrapper to swap the arguments.

Reply:
I guess this is moving the wrapper down one level.

Reply:
My recollection was that Seq had an internal wrapper that flipped the arguments

Reply:
I couldn't see such a wrapper, but maybe I missed it?

Reply:
https://github.com/creusot-rs/creusot/blob/master/creusot-contracts/src/logic/seq.rs#L48 ?

Reply:
Ah i guess i only did it for sets: https://github.com/creusot-rs/creusot/blob/master/creusot-contracts/src/logic/fset.rs#L50-L52

Reply:
we're not 100% consistent either way, I have a slight preference for the rust wrapper approach but i wouldn't consider it blocking.

Reply:
I agree that it is better. I pushed a new version with a rust wrapper instead of one in why3